### PR TITLE
add recipes for theseus and mmligner

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

@tiggha, @Koesed96, please post a reply here to confirm you want to be co-maintainers of these packages! And thanks for your initial work on this!

### CODE REVIEW

**Comment by jaimergp** [comment]
Dear members of @conda-forge/help-c-cpp, I think this is ready for review. I tried to build for Windows too but there's too many UNIX-isms to patch. If anything, we can try at some point in the future in the feedstock. For now we will be happy with Linux and MacOS.

**Comment by Koesed96** [comment]
> Checklist
> 
> * [x]  Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
> * [x]  License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
> * [x]  Source is from official source
> * [x]  Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
> * [ ]  If static libraries are linked in, the license of the static library is packaged.
> * [x]  Build number is 0
> * [x]  A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
> * [ ]  GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
> 
> @Tiggha, @Koesed96, please post a reply here to confirm you want to be co-maintainers of these packages! And thanks for your initial work on this!

Yes, I would like to be a co-maintainer for these packages!

**Comment by Tiggha** [comment]
> 
> 
> Checklist
> 
>     * [x]  Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
> 
>     * [x]  License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
> 
>     * [x]  Source is from official source
> 
>     * [x]  Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
> 
>     * [ ]  If static libraries are linked in, the license of the static library is packaged.
> 
>     * [x]  Build number is 0
> 
>     * [x]  A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
> 
>     * [ ]  GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
> 
> 
> @Tiggha, @Koesed96, please post a reply here to confirm you want to be co-maintainers of these packages! And thanks for your initial work on this!

Yes, I would like to be a co-maintainer for these packages!

**Comment by synapticarbors** [comment]
Is this PR meant to replace https://github.com/conda-forge/staged-recipes/pull/11256?

**Comment by jaimergp** [comment]
Oops, sorry, yes, completely forgot about that one.

**Comment by jaimergp** [comment]
I will apply your suggestions there here. Sorry for the inconvenience!

File: recipes/mmligner/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  sha256: 0208e8fd1eb0492a13d1bfcba53a7a97f96023f18e3982e334962f0ce1c69f07
+  patches:
+    - add_cli_h_flag.patch
+    - makefile.patch
+
+build:
+  number: 0
+  script: make && mkdir -p ${PREFIX}/bin && cp bin/mmligner ${PREFIX}/bin/mmligner  # [unix]
+  skip: True  # [win]
+  # script: jom && mkdir "%SCRIPTS%" && copy "bin\mmligner.exe" "%SCRIPTS%\mmligner.exe"  # [win]
```
Comment (marcelotrevisani) [inline recipes/mmligner/meta.yaml:19]:
```suggestion
```

File: recipes/mmligner/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  number: 0
+  script: make && mkdir -p ${PREFIX}/bin && cp bin/mmligner ${PREFIX}/bin/mmligner  # [unix]
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make  # [unix]
+    - jom  # [win]
+    - git  # [win]
```
Comment (marcelotrevisani) [inline recipes/mmligner/meta.yaml:25]: Why do you need ``git`` for windows?

Comment (jaimergp) [reply recipes/mmligner/meta.yaml:25]: It included `sed`, but now we are not providing any Windows packages here (for now) so we can just delete these lines! Thanks for the reminder!

**Pull request merged**